src/cubic_loader/qlik/utils.py

```diff
@@ -267,3 +268,16 @@ def dataframe_from_merged_csv(csv_path: str, dfm_path: str) -> pl.DataFrame:
     schema = polars_schema_from_dfm(dfm_path)
     df = pl.read_csv(csv_path, schema=schema).filter(pl.col("header__change_oper").ne("B"))
     return df
+
+
+def key_column_join_type(df: pl.DataFrame, key_columns: List[str]) -> List[Tuple[str, str]]:
+    """
+    Check for NULL counts in key_columns to determine if `=` or `IS NOT DISTINCT FROM` can be used
+    """
+    return_list = []
+    for column in key_columns:
+        if df.get_column(column).null_count() > 0:
```

Comment:
Any risk here with a file with many records here? Polars is pretty fast so I'm assuming not.

Reply:
No, we're doing all the existing operations in polars already. We limit the "csv" size of each batch to 60mb, so the memory footprint for any batch is always less than 1gb.

Reply:
Makes sense. Thanks for the extra explanation.